src/serena/util/file_system.py

```diff
@@ -1,20 +1,18 @@
 import os
-from collections.abc import Sequence
+from collections.abc import Callable
 
 
 def scan_directory(
     path: str,
     recursive: bool = False,
     relative_to: str | None = None,
-    ignored_dirs: Sequence[str] = (),
-    ignored_files: Sequence[str] = (),

```

Comment:
should we not also have is_ingored_file here? If feel like it would be more convenient than having to filter afterwards